compiler/rustc_hir_typeck/src/method/mod.rs

```diff
@@ -212,7 +212,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             let mut candidates =
                 match self.lookup_probe(segment.ident, self_ty, call_expr, ProbeScope::AllTraits) {
                     // If we find a different result the caller probably forgot to import a trait.
-                    Ok(ref new_pick) if *new_pick != pick => {
+                    Ok(ref new_pick) if new_pick.self_ty != pick.self_ty => {
```

Comment:
can you not have different traits with the same `self_ty` here? I would expect us to also have to check the `Item` here

Reply:
Yea... I need to dig into why [doing that](https://github.com/rust-lang/rust/pull/104394/commits/fb1a71a985c3b040e114ec1dc96a4ebe8360718f) causes a diagnostic regression